src/net.rs

```diff
@@ -240,7 +283,10 @@ pub fn test_read_commands(){
     // read the command from the stream for Command::Query("select")  
     command_res = read_commands(&mut Cursor::new(vec2));
     assert_eq!(command_res.is_ok(), true);
-    assert_eq!(command_res.unwrap(), Command::Query("select".into()));
+    match command_res.unwrap() {
+        Command::Query(query) => assert_eq!(query, "select"),
+        _ => assert!(false)
+    }
```

Comment:
Instead of `assert!(false)` you  should maybe use `panic!("short description why the test failed");`